### ISSUE TITLE
IIFE in attributes get incorrectly unwrapped

### ISSUE THREAD

**Issue by edemaine** [ryansolid/dom-expressions#335]
[Playground link](https://playground.solidjs.com/anonymous/1d9fc76a-0960-4c30-8bb0-fdc06884bbff)

## Input

```tsx
<For
  each={((s, e) => {
    let step = e > s ? 1 : -1;
    return Array.from(
      { length: Math.abs(e - s) },
      (_, i) => s + i * step,
    );
  })(0, params.boxes)}
/>
```

(this code may look a bit weird, but it's [generated](https://civet.dev/playground?code=PEZvciBlYWNoPVswLi4ucGFyYW1zLmJveGVzXT4%3D) from the simple Civet code `<For each=[0...params.boxes]>`)

## Output

```js
import { createComponent as _$createComponent } from "solid-js/web";
import { For as _$For } from "solid-js/web";
_$createComponent(_$For, {
  get each() {
    let step = e > s ? 1 : -1;
    return Array.from({
      length: Math.abs(e - s)
    }, (_, i) => s + i * step);
  }
});
```

The bindings for `s` and `e` (by the IIFE) got lost. ☹️ 

This bug seems to go back to at least Solid 1.8.5.

## Workaround

Wrap attribute in another IIFE: `() => ...`

**Comment by ryansolid** [comment]
Yeah that ones a bit odd.. I wouldn't expect the iife to get swallowed since we usually check for argument length, and that clearly has arguments.

**Comment by edemaine** [comment]
This seems to be fixed by #327. So maybe just needs a release.

**Comment by ryansolid** [comment]
That was released like 7 versions ago. 0.37.22... we are on 0.38.5. The release should have gone out June 25th. Can you confirm if you have it?
